src/main/java/com/salesforce/dynamodbv2/mt/mappers/CreateTableRequestBuilder.java

```diff
@@ -47,7 +47,7 @@ public CreateTableRequest build() {
 
     private void setDefaults() {
         if (createTableRequest.getProvisionedThroughput() == null && (createTableRequest.getBillingMode() == null
-                || createTableRequest.getBillingMode().equals(BillingMode.PROVISIONED))) {
+                || createTableRequest.getBillingMode().equals(BillingMode.PROVISIONED.name()))) {
```

Comment:
@degaydorus ... This was caught by the "equals() between objects of inconvertible types" check.  It looks like it was a potential bug.  I think it may have worked somewhat by coincidence.  Can you confirm?

Reply:
Thanks for sharing, it is a bug and the fix you added it's what's needed.  We didn't see it live because this case is uncommon.  Here's a PR off of code-cleanup branch that adds a repro test as well as some additional tests & cleanup: https://github.com/salesforce/mt-dynamo/pull/335

Reply:
Cool.  I merged it.